exa_error.go

```diff
@@ -0,0 +1,75 @@
+package error_reporting_go
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+type ErrorMessageBuilder struct {
+	errorCode  string
+	message    string
+	parameters [][]string
+	mitigation string
+}
+
+func ExaError(errorCode string) *ErrorMessageBuilder {
+	return &ErrorMessageBuilder{
+		errorCode: errorCode,
+	}
+}
+
+func (builder *ErrorMessageBuilder) Message(message string) *ErrorMessageBuilder {
+	builder.message = message
+	return builder
+}
+
+func (builder *ErrorMessageBuilder) Parameter(arguments ...interface{}) *ErrorMessageBuilder {
+	validateArguments(arguments)
+	builder.parameters = append([][]string{{fmt.Sprintf("%v", arguments[0]), fmt.Sprintf("%v", arguments[1])}})
+	return builder
+}
+
+func (builder *ErrorMessageBuilder) Mitigation(mitigation string) *ErrorMessageBuilder {
+	builder.mitigation = mitigation
+	return builder
+}
+
+func validateArguments(arguments []interface{}) {
+	if len(arguments) < 2 || len(arguments) > 3 {
+		panic("Parameter function accepts 2 or 3 arguments: parameter name, parameter value, (optional) description")
+	}
+	if getInterfaceType(arguments[0]) != "string" {
+		panic("Parameter function's first argument 'parameter name' must be a string")
+	}
+	if len(arguments) == 3 && getInterfaceType(arguments[2]) != "string" {
+		panic("Parameter function's third argument 'description' must be a string")
+	}
+}
```

Comment:
In the other error-reporting-libs we have the concept to always try to make the best out of it and never fail.
Reason: If something is wrong with the exception message the crawler should catch that.
If it does not catch it, customers should better see a broken error message (for example with one parameter not filled) than an error message, that the error message is broken.
In that case you could log a warning and not add the parameter.
In the other libs we display a missing parameter as something like `!!!MISSING PARAMETER!!!`.